pkg/resource/deploy/step_generator.go

```diff
@@ -555,7 +555,83 @@ func (sg *stepGenerator) GenerateDeletes() []Step {
 			}
 		}
 	}
-	return dels
+
+	if len(targets) > 0 {
+		logging.V(7).Infof("Planner was asked to only delete '%v'", targets)
+
+		resourcesToDelete := make(map[resource.URN]bool)
+
+		// Do an initial pass first to ensure that all the targets mentioned are ones we know about.
+		for _, target := range targets {
+			if _, has := sg.plan.olds[target]; !has {
+				logging.V(7).Infof("Resource to delete (%v) could not be found in the stack.", target)
+				if strings.Contains(string(target), "$") {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundError(), target)
+				} else {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundDidYouForgetError(), target)
+				}
+				return nil, result.Bail()
```

Comment:
This is still failing before we've checked all the targets--kind of think it'd be nice to list all missing targets before we bail.

Reply:
Oh, i see.  Sure, i can add that information.

Reply:
Done.